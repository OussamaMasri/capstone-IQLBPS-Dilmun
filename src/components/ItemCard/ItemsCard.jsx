import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

function ItemsCard({ item, animate }) {
  const { t } = useTranslation();
  return (
    // keep animate prop false if you don't want the slide animation
    <div
      data-testid="item-card"
      className={`lg:w-80 w-auto bg-background ${
        animate ? 'lg:animate-scroll' : ''
      } m-7 rounded-lg hover:shadow-secondary shadow-gray-400 shadow-lg hover:shadow-2xl duration-300`}
    >
      <Link to={`/product/${item.id}`}>
        <figure className="relative max-w-sm transition-all cursor-pointer">
          <img
            className="rounded-t-lg overflow-hidden object-cover mb-3 lg:h-80 h-80 shadow-md xl:w-full w-80 "
            src={item.file}
            alt="product"
          />
          <figcaption className="absolute top-3 right-1 px-4 text-lg">
            {item.type === 'Donated' ? (
              <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                {item.type}
              </span>
            ) : (
              <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                {item.price} $
              </span>
            )}
          </figcaption>
        </figure>
        <div className="px-5 pb-5">
          {item.title.length > 15 ? (
            <h5 className="text-3xl pb-2 font-extrabold text-primary">
              {item.title.slice(0, 1).toUpperCase()}
              {item.title.slice(1, 15)} ...
            </h5>
          ) : (
            <h5 className="text-3xl pb-2 font-extrabold text-primary">
              {item.title.slice(0, 1).toUpperCase()}
              {item.title.slice(1)}
            </h5>
          )}
          <span className="text-xl py-1 font-semibold text-primary">
            {item.type}
          </span>
          {item.description.length > 20 ? (
            <h5 className="text-base py-1 font-base text-primary">
              {item.description.slice(0, 1).toUpperCase()}
              {item.description.slice(1, 30)}...
            </h5>
          ) : (
            <h5 className="text-base py-1 font-base text-primary">
              {item.description.slice(0, 1).toUpperCase()}
              {item.description.slice(1)}
            </h5>
          )}
          <div className="">
            {typeof item.category === 'object' ? (
              item.category.slice(0, 3).map((cat) => {
                return (
                  <span
                    key={cat}
                    className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg"
                  >
                    {cat}
                  </span>
                );
              })
            ) : (
              <span className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg">
                {item.category}
              </span>
            )}
            <h5 className="text-base py-1 font-base text-primary">
              {item.createdAt}
            </h5>
          </div>
          <button
            type="button"
            className="w-full my-2 duration-300 text-background bg-secondary hover:bg-primary focus:ring-1 focus:outline-none focus:ring-tertiary font-medium rounded-lg text-md px-5 py-2.5 text-center"
          >
            <a href={`mailto: ${item.owner?.email}`}>{t('email_owner')}</a>
          </button>
        </div>
      </Link>
    </div>
  );
}
export default ItemsCard;
