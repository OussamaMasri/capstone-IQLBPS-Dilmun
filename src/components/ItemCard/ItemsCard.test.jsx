import { render, screen, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import ItemsCard from './ItemsCard';
import { store } from '../../app/store';

const MOCK_ITEM_DATA = {
  title: 'Title',
  file: 'https://cdn.discordapp.com/attachments/1031834305703460906/1041710013992947812/image.png',
  description: 'Description',
  price: 110,
  type: 'Item Type',
  categories: ['Men'],
  owner: {
    email: 'email',
  },
};
afterEach(cleanup);

test('render item cards correctly', () => {
  render(
    <BrowserRouter>
      <Provider store={store}>
        <ItemsCard item={MOCK_ITEM_DATA} />
      </Provider>
    </BrowserRouter>
  );

  expect(screen.getByTestId('item-card')).toMatchSnapshot();
});
