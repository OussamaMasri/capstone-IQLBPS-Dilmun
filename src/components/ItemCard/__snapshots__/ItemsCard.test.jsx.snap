// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render item cards correctly 1`] = `
<div
  class="lg:w-80 w-auto bg-background  m-7 rounded-lg hover:shadow-secondary shadow-gray-400 shadow-lg hover:shadow-2xl duration-300"
  data-testid="item-card"
>
  <a
    href="/product/undefined"
  >
    <figure
      class="relative max-w-sm transition-all cursor-pointer"
    >
      <img
        alt="product"
        class="rounded-t-lg overflow-hidden object-cover mb-3 lg:h-80 h-80 shadow-md xl:w-full w-80 "
        src="https://cdn.discordapp.com/attachments/1031834305703460906/1041710013992947812/image.png"
      />
      <figcaption
        class="absolute top-3 right-1 px-4 text-lg"
      >
        <span
          class="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2"
        >
          110
           $
        </span>
      </figcaption>
    </figure>
    <div
      class="px-5 pb-5"
    >
      <h5
        class="text-3xl pb-2 font-extrabold text-primary"
      >
        T
        itle
      </h5>
      <span
        class="text-xl py-1 font-semibold text-primary"
      >
        Item Type
      </span>
      <h5
        class="text-base py-1 font-base text-primary"
      >
        D
        escription
      </h5>
      <div
        class=""
      >
        <span
          class="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg"
        />
        <h5
          class="text-base py-1 font-base text-primary"
        />
      </div>
      <button
        class="w-full my-2 duration-300 text-background bg-secondary hover:bg-primary focus:ring-1 focus:outline-none focus:ring-tertiary font-medium rounded-lg text-md px-5 py-2.5 text-center"
        type="button"
      >
        <a
          href="mailto: email"
        >
          email_owner
        </a>
      </button>
    </div>
  </a>
</div>
`;
